fedimint-core/src/api.rs

```diff
@@ -801,12 +801,21 @@ impl WsAuthenticatedApi {
         }
     }
 
+    pub async fn set_password(&self, auth: ApiAuth, key: DecryptionKey) -> FederationResult<()> {
+        self.request("/set_password", ApiRequestErased::new((auth, key)))
+            .await
+    }
+
```

Comment:
It would be the same user-created password, but the plan is the plaintext password would never touch `fedimintd` but be hashed client-side to derive the auth token and the decryption key.

Reply:
> This gives clients some flexibility in choosing how to stretch the key

Flexibility is typically not what you want from a cryptographic protocol if you can avoid it. If there are to be multiple ways these should be enumerated in a standards document (so different client impls can follow a common set) and the fedimint instance should tell clients which way to use on setup to avoid a scenario like with WiFis where it's "good luck guessing which combination of options you need to make this enterprise WiFi work even though you know your login data".

Reply:
Currently the requirement is it is in PHC format which is the closest thing to a standard for hashed passwords.

The server-side doesn't necessarily need to care about how the password was hashed, but I guess if there are multiple clients that need to be compatible they should use the same hashing algorithm and salt.